### PR TITLE
B 20493 int

### PR DESCRIPTION
**Pull request by deandreJones**
## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the Office app
2. Click Request Account 
3. For this test- we want to be able to select all available transportation offiices- even those that don't office ppm_closeout- , JPPSO Southeast (CNNQ) is one example, where before this change- it was not able to be selected, after this change it is able to be selected

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

## Screenshots

image

### CODE REVIEW

File: pkg/services/transportation_office/transportation_office_fetcher.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func FindTransportationOffice(appCtx appcontext.AppContext, search string) (mode
 	sqlQuery := `
 	with names as (select office.id as transportation_office_id, office.name, similarity(office.name, $1) as sim
         from transportation_offices as office
-        where name % $1 and provides_ppm_closeout is true
+        where name % $1
 		order by sim desc
         limit 5)
 select office.*
```
Comment (danieljordan-caci) [inline pkg/services/transportation_office/transportation_office_fetcher.go:70]: So I know this works, but I have some reservations.

This is also used for customers when searching for transpo offices in `/pkg/handlers/internalapi/transportation_offices.go:54`:
```
transportationOffices, err := h.TransportationOfficesFetcher.GetTransportationOffices(appCtx, params.Search)
			if err != nil {
				appCtx.Logger().Error("Error searching for Transportation Offices: ", zap.Error(err))
				return transportationofficeop.NewGetTransportationOfficesInternalServerError(), err
			}
```

and...

This is also used for customers when searching for transpo offices in `/pkg/handlers/ghcapi/transportation_offices.go:23`:
```
transportationOffices, err := h.TransportationOfficesFetcher.GetTransportationOffices(appCtx, params.Search)
			if err != nil {
				appCtx.Logger().Error("Error searching for Transportation Offices: ", zap.Error(err))
				return transportationofficeop.NewGetTransportationOfficesInternalServerError(), err
			}
```

So I think that if that is fine for customers and office users inside of the app to see all transpo offices, then we can leave this.

If not, then we should probably create a new service object that only this open handler uses. Should prob ask Dan.

**Review by danieljordan-caci — CHANGES_REQUESTED** [review]
Need you to clarify some things with Dan before signing off or doing additional work.

File: pkg/services/transportation_office/transportation_office_fetcher.go (hunk 1 above)
Comment (deandreJones) [reply pkg/services/transportation_office/transportation_office_fetcher.go:70]: good call

Comment (deandreJones) [reply pkg/services/transportation_office/transportation_office_fetcher.go:70]: after some research- this endpoint can't be used, will have to create anew one or- use the admin api

image

Comment (danieljordan-caci) [reply pkg/services/transportation_office/transportation_office_fetcher.go:70]: Can just create a new service object that the handler calls. Just copy over what's in `pkg/services/transportation_office/transportation_office_fetcher.go`:

```
func (o transportationOfficesFetcher) GetTransportationOfficesForRequestedOfficeUsers(appCtx appcontext.AppContext, search string) (*models.TransportationOffices, error) {
	officeList, err := FindTransportationOfficeForRequestedOfficeUsers(appCtx, search)

	if err != nil {
		switch err {
		case sql.ErrNoRows:
			return &officeList, apperror.NewNotFoundError(uuid.Nil, "Search string: "+search)
		default:
			return &officeList, err
		}
	}

	return &officeList, nil
}

func FindTransportationOfficeForRequestedOfficeUsers(appCtx appcontext.AppContext, search string) (models.TransportationOffices, error) {
	var officeList []models.TransportationOffice

	// The % operator filters out strings that are below this similarity threshold
	err := appCtx.DB().Q().RawQuery("SET pg_trgm.similarity_threshold = 0.03").Exec()
	if err != nil {
		return officeList, err
	}

	sqlQuery := `
	with names as (select office.id as transportation_office_id, office.name, similarity(office.name, $1) as sim
        from transportation_offices as office
        where name % $1
		order by sim desc
        limit 5)
select office.*
        from names n inner join transportation_offices office on n.transportation_office_id = office.id
        group by office.id
        order by max(n.sim) desc, office.name
        limit 5`
	query := appCtx.DB().Q().RawQuery(sqlQuery, search)
	if err := query.All(&officeList); err != nil {
		if errors.Cause(err).Error() != models.RecordNotFoundErrorString {
			return officeList, err
		}
	}
	for i := range officeList {
		err := appCtx.DB().Load(&officeList[i], "Address")
		if err != nil {
			return officeList, err
		}
	}
	return officeList, nil
}
```

Bing bang boom.

Comment (deandreJones) [reply pkg/services/transportation_office/transportation_office_fetcher.go]: I don't like copied code- I prefer to make the code more flexible for multiple use

**Comment by danieljordan-caci** [comment]
You just not putting labels on your PRs now, boss? :)

**Review by traskowskycaci — APPROVED** [review]
Query results look good!

May need a `make mocks_generate` on your branch

**Comment by deandreJones** [comment]
> You just not putting labels on your PRs now, boss? :)

I never claimed to be perfect...

**Comment by deandreJones** [comment]
> Query results look good!
> 
> May need a `make mocks_generate` on your branch

already did

File: pkg/services/transportation_office/transportation_office_fetcher.go (hunk 1 above)
Comment (danieljordan-caci) [reply pkg/services/transportation_office/transportation_office_fetcher.go]: Fine by me! Your solution looks good!

**Review by danieljordan-caci — APPROVED** [review]
Nailed it!

**Pull request merged**